src/bin/pg_dump/pg_dump_segment_helper.c

```diff
@@ -0,0 +1,219 @@
+/*-------------------------------------------------------------------------
+ *
+ * pg_dump_segment_helper.c
+ *	  pg_dump_segment_helper provides functions which saves data from stdin
+ *	  to files and restores data from files to stdout
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "parallel.h"
+
+#include <ctype.h>
+
+#include "pg_backup.h"
+#include "pg_backup_archiver.h"
+#include "pg_backup_utils.h"
+#include "pg_dump_segment_helper.h"
+
+static int runRestore_Archive(const char *inputFileSpec);
+static int runRestore_PlainText(const char *inputFileSpec);
+#ifdef HAVE_LIBZ
+static int runRestore_CompressedPlainText(const char *inputFileSpec);
+#endif
+
+static void setupDumpWorkerDummy(Archive *AH, RestoreOptions *ropt);
+
+static void addDumpableObject(Archive *AH);
+static void ConfigureForClearData(Archive *AH);
+static int	getDumpFromStdin(Archive *AH, void *dcontext);
+
+int
+onSegmentHelper_RunDump(const char *inputFileSpec, int compressLevel, ArchiveFormat format)
+{
+	Archive		   *AH;
+	RestoreOptions *ropt;
+
+	int plainText = 0;
+
+	/* archiveFormat specific setup */
+	if (format == archNull)
+		plainText = 1;
+
+	/* Open the output file */
+	AH = CreateArchive(
+		inputFileSpec, format, compressLevel, archModeWrite, setupDumpWorkerDummy
+	);
+
+	/* Register the cleanup hook */
+	on_exit_close_archive(AH);
+
+	if (AH == NULL)
+		exit_horribly(NULL, "could not open output file \"%s\" for writing\n", inputFileSpec);
+
+	ConfigureForClearData(AH);
+
+	addDumpableObject(AH);
+
+	/*
+	 * Set up options info to ensure we dump what we want.
+	 */
+	ropt = NewRestoreOptions();
+	ropt->filename = inputFileSpec;
+
+	if (compressLevel == -1)
+		ropt->compression = 0;
+	else
+		ropt->compression = compressLevel;
+
+	((ArchiveHandle *)AH)->ropt = ropt;
+
+	if (plainText)
```

Comment:
Can we use `format == archNull` in this condition and remove the plainText variable?

Reply:
make sense, done